src/main/resources/META-INF/jqassistant-rules/transaction/jee-transaction.xml

```diff
@@ -3,27 +3,32 @@
                    xsi:schemaLocation="http://schema.jqassistant.org/rule/v2.2 https://jqassistant.github.io/jqassistant/current/schema/jqassistant-rule-v2.2.xsd">
 
     <group id="jee-transaction:Default">
-        <includeConstraint refId="jee-transaction:TransactionalMethodMustNotBeInvokedFromSameClass"/>
+        <includeConstraint refId="jee-transaction:TransactionalMethodMustNotBeInvokedFromSameClassOrSubclass"/>
         <includeConstraint refId="jee-transaction:PrivateMethodMustNotBeAnnotatedWithTransactional"/>
     </group>
 
-    <constraint id="jee-transaction:TransactionalMethodMustNotBeInvokedFromSameClass">
+    <constraint id="jee-transaction:TransactionalMethodMustNotBeInvokedFromSameClassOrSubclass">
         <requiresConcept refId="jee-transaction:TransactionalMethod"/>
         <requiresConcept refId="java:GeneratedType"/>
+        <requiresConcept refId="java:MemberInheritedFrom"/>
         <description>Transactional methods must not be invoked from the same class.</description>
         <cypher><![CDATA[
             MATCH
-              (artifact:Artifact)-[:CONTAINS]->(type:Type)-[:DECLARES]->(calledMethod:Method:JEE:Transactional),
-              (type:Type)-[:DECLARES]->(callingMethod:Method),
-              (callingMethod:Method)-[invokes:INVOKES]->(calledMethod)
+              (artifact:Artifact)-[:CONTAINS]->(callingType:Java:Type),
+              (callingType)-[:EXTENDS|IMPLEMENTS*0..]->(:Java:Type)-[:DECLARES]->(calledMethod:Method:JEE:Transactional),
```

Comment:
Looks a bit strange: in a short form this line says `callingType declares calledMethod` - is this by intention?